apps/wing/src/analytics/collectors/ci-collector.ts

```diff
@@ -0,0 +1,76 @@
+import { Collector } from "./collector";
+
+export interface CIData {
+  name: string;
+}
+
+export enum CIType {
+  GITHUB_ACTIONS = "github_action",
+  GITLAB_CI = "gitlab_ci",
+  JENKINS = "jenkins",
+  CIRCLECI = "circleci",
+  BITBUCKET = "bitbucket",
+  AZURE_DEVOPS = "azure_devops",
+  TEAMCITY = "teamcity",
+  AWS_CODEBUILD = "aws_codebuild",
+  NONE = "none",
+}
+
+export class CICollector extends Collector {
+  
```

Comment:
Nit: This class feels a bit over designed...

I'd just write it like so:

```js
// list of supported CI environment variables
const ENV = [
  "GITHUB_ACTIONS",
  "GITLAB_CI",
  // ...
};

export class CICollector extends Collector {
  async collect(): Promise<CIData | undefined> {
    for (const e of ENV) {
      if (e in process.env) {
        return { name: e };
      }
    }

    return undefined;
  }
}
```

I don't see a real need to mangle the env name in the `name` and also, we can update the contract of the `Collector` to allow `undefined` to be returned, indicating this collection is empty.

(this requires a change to `collect.ts` to not call `canCollect()`.

Reply:
thats fair, yea this originally was a lot more complicated when I had it designed to get much more information than just the CI environment name. (repo id, commit hash, etc..) Ill simplify this

Reply:
Although with mangling the name, its not always the case that an environment id is a nice indicator of what CI it is in. Like with Jenkins we only know based on JENKINS_URL which is not really a good name, but we would know its Jenkins (which could be okay) but a step worse might be azure which is identified by the environment variable `BUILD_BUILDID` I would have no idea looking at downstream analytics that is azure

Reply:
So you can make `ENV` a map from variable to identifier.